Default: Allow papyrus growing on default:sand

paramat opened the pull request:
Also:
Mgv5/mgv7 papyrus schematic now has sand nodes as a base
Add beaches of default:sand to mgv5/mgv7 deserts

![screenshot_20150830_001944](https://cloud.githubusercontent.com/assets/3686677/9564888/9a580926-4ead-11e5-81e1-1eadfabe7c39.png)

^ This commit

![screenshot_20150830_002323](https://cloud.githubusercontent.com/assets/3686677/9564890/a03e6be6-4ead-11e5-92d5-c196fbdcb82e.png)

^ Currently

Sand base nodes blend with beaches and shallow water, it looks better.
In mgv5/mgv7 papyrus grows in shallow water in savanna and desert biomes, allowing papyrus to grow on sand allows growing on the beaches of these biomes without having to place dirt there first. In reality papyrus can grow in wet sandy soil.
The sand around the papyrus base can be considered sand that has got caught-up in, and collected around, the base of the papyrus.

kilbith commented on mods/default/functions.lua:
```diff
@@ -152,7 +152,8 @@ end
 	local name = minetest.get_node(pos).name
-	if name ~= "default:dirt_with_grass" and name ~= "default:dirt" then
+	if name ~= "default:dirt_with_grass" and name ~= "default:dirt" and
+			name ~= "group:sand" then
```
``` Lua
if not name:find("dirt") and not name:find("sand") then
```

sfan5 replied:
* We don't want papyrus to grow on <i>sand</i>worm_spawners or on stuff registered by a tool<i>sand</i>blocks mod. Yes I made those examples up but we didn't ever do this in any other code, why do it here?

* Does this actually work? I don't think the name will be "group:sand" if it's in the sand group. Don't you need to look up the table of registered nodes for that?

paramat commented:
Updated. Thanks for catching 'name' not working with 'group:sand'.
I changed the added fertile node to be default:sand, assumed to be wet sand near water.
I also added beaches of default:sand to desert biomes so that papyrus can be grown there.

sfan5 commented:
:+1:

paramat commented:
4d9b595a69988bcd9add832ba4289c5e8285c800
